docs/fri.md

```diff
@@ -1 +1,137 @@
-# FRI
+# Fast Reed - Solomon Interactive Oracle Proof of Proximity
+
+References:
+
++ [Low Degree Testing](https://medium.com/starkware/low-degree-testing-f7614f5172db)
++ [STARKs by Vitalik](https://vitalik.eth.limo/general/2017/11/22/starks_part_2.html)
++ [FRI Mechanics by RISC Zero](https://www.youtube.com/watch?v=wqRuoyH3Mqk&t=1032s&ab_channel=RISCZero)
++ [FRI explanation by example](https://dev.risczero.com/proof-system/stark-by-hand#lesson-11-fri-protocol-commit-phase)
++ [FRI lecture](https://www.youtube.com/watch?v=A3edAQDPnDY&ab_channel=Blockchain-Web3MOOCs)
+
+## Low Degree Testing
+
+Low degree testing refers to the problem of deciding whether a given function is a polynomial of some bounded degree.
+
+FRI protocol uses the idea of testing two polynomials with half the queries and splitting a polynomial into two smaller ones.
+
+## What is FRI For?
+
+Proving a vector commitment is (close to) a Reed-Solomon codeword (Low-degree polynomial).
+
+## How Does FRI Work?
+
+Iteratively "folding" the commitment into smaller and smaller commitments.
+
+Queries ensure the folding was done properly.
+
+## Roadmap
+
+1. [Merkle Tree](fri.md#Merkle%20Tree)
+2. [Mechanics of FRI](fri.md#Mechanics%20of%20FRI)
+3. [Polynomial Commitment from FRI](fri.md#Polynomial%20Commitment%20from%20FRI)
+
+## Merkle Tree
+
+FRI ultilizes Merkle trees to commit to a vector. Each leaf of the trees represents an entry in the
+vector, with the root serving as the commitment.
+
+![merkle_tree.png](attachments/merkle_tree.png)
+
+A commitment to a polynomial $q$, where each leaf corresponds to an evaluation of $q$ at a specific point.
+
+The prover affirms that the leaves represent evaluations of $q$.
+
+## Mechanics of FRI
+
+### Prequirisites Knowledge
+
+Prover Merkle-commits to evaluations $q(x)$ for those $x$ in a subset $\ohm$ of $\mathbb{F}_p$.
+
+Subset $\ohm$ has size $n = \rho^{-1}k$, comprises all $n$-th [roots of unity](../docs/plonk.md#Roots%20of%20Unity) in $\mathbb{F}_p$.
+
++ $k$ is the degree of $q$.
++ Assume $n$ is a power of 2.
++ $\rho^{-1} \geq 2$ is called the "FRI blowup factor".
++ $\rho \leq 1/2$ is called the "rate of the Reed-Solomon code" used.
++ $\lambda$ is the security parameter a.k.a "$\lambda$ bits of security".
```

Comment:
We have the description of $\lambda$ here but i don't see the formula with $\lambda$ anywhere

Reply:
I mentioned it in the query phase section